keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Prerequisite testing updates
+
+#### Unit tests
+
+Components in the architecture presented above are already existing (and
+supposedly are already unit tested): kube-proxy's healthz server, kube-proxy's
+"ETP=local" server. We would need to add tests which validate the change in
+behaviour for opening up additional ports for ETP=cluster services and validate
+that the difference in HC logic between ETP services is captured correctly.
+
+#### Integration tests
+
+Kube-proxy is mostly tested via unit tests and e2e, not integration, and this is
+not expected to change.
+
+#### e2e tests
+
+Kubernetes in general needs to extended its load balancing test suite with
+disruption tests, this might be the right effort we need to get that ball
+rolling. Testing would include: validation that HC probe's return a respond
+corresponding to the "state of the Kubernetes cluster world" during different
+disruption scenarios.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature flag for changing the API for ETP=cluster services and
+  include `healthCheckNodePort` for them. This will help testing and graduation
+  to beta and stable.  
+- Changes to data plane components get implemented (Kube-proxy)
+- Changes to control plane components get implemented, but depends on the
+  feature flag being enabled or not.
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Documentation is changed for ETP=cluster services to highlight the change with
+  regards to `healthCheckNodePort`.
+- Testing of Kubernetes with the feature flag enabled is green and reports no
+  major incidents.
+
+#### GA
+
+- Version skew delta has passed.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+The possible version skew between the control and data plane in a Kubernetes
```

Comment:
version skew is more about I have 2 control planes, a and b, and one has version 1.X and the other 1.X+1.

Depending on where the request land it will create the class A service with HCNodePort or no

Reply:
> version skew is more about I have 2 control planes, a and b, and one has version 1.X and the other 1.X+1.

That seems strange to me. Are there clusters running with two control planes with different versions, _at the same time_ and which manage the same data plane? How is that managed then?

Reply:
when you  upgrade the apiservers, you do it one at a time to have full availability, in a cluster with 3 apiserves A,B and C, you upgrade A until it is with version X+1 and then you do the same sequentially for the others